segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/filesystem/FileSystemStorage.java

```diff
@@ -0,0 +1,377 @@
+                    !(attrs.permissions().contains(OWNER_WRITE)), false,
+                    new ImmutableDate(attrs.creationTime().toMillis()));
+
+            return information;
+    }
+
+    private boolean syncExists(String streamSegmentName) {
+        boolean exists = Files.exists(Paths.get(config.getFilesystemRoot(), streamSegmentName));
+        return exists;
+    }
+
+    @SneakyThrows
+    private SegmentProperties syncCreate(String streamSegmentName) {
+        log.info("Creating Segment {}", streamSegmentName);
+            Set<PosixFilePermission> perms = new HashSet<>();
+            // add permission as rw-r--r-- 644
+            perms.add(PosixFilePermission.OWNER_WRITE);
+            perms.add(PosixFilePermission.OWNER_READ);
+            perms.add(PosixFilePermission.GROUP_READ);
+            perms.add(PosixFilePermission.OTHERS_READ);
+            FileAttribute<Set<PosixFilePermission>> fileAttributes = PosixFilePermissions.asFileAttribute(perms);
+
+            Path path = Paths.get(config.getFilesystemRoot(), streamSegmentName);
+            Files.createDirectories(path.getParent());
+            Files.createFile(path, fileAttributes);
+            log.info("Created Segment {}", streamSegmentName);
+            return this.getStreamSegmentInfo(streamSegmentName, null).get();
+    }
+
+    @SneakyThrows
+    private Void syncWrite(SegmentHandle handle, long offset, InputStream data, int length) {
+        log.trace("Writing {} to segment {} at offset {}", length, handle.getSegmentName(), offset);
+        Path path = Paths.get(config.getFilesystemRoot(), handle.getSegmentName());
+
+        if (handle.isReadOnly()) {
+            log.warn("Write called on a readonly handle of segment {}", handle.getSegmentName());
+            throw new IllegalArgumentException("Write called on a readonly handle of segment "
+                    + handle.getSegmentName());
+        }
+
+        //Fix for Jenkins as Jenkins runs as super user privileges.
+        if ( !isWritableFile(path)) {
+            throw new StreamSegmentSealedException(handle.getSegmentName());
+        }
+
+        long fileSize = path.toFile().length();
+        if (fileSize < offset) {
+            throw new BadOffsetException(handle.getSegmentName(), fileSize, offset);
+        } else {
+            try (FileChannel channel = FileChannel.open(path,
+                    StandardOpenOption.WRITE); ReadableByteChannel sourceChannel = Channels.newChannel(data)) {
+                while (length != 0) {
+                    long bytesWritten = channel.transferFrom(sourceChannel, offset, length);
+                    offset += bytesWritten;
+                    length -= bytesWritten;
+                }
+            }
+            return null;
+        }
+    }
+
+    private boolean isWritableFile(Path path) throws IOException {
+        PosixFileAttributes attrs = null;
```

Comment:
1. Indentation.
2. You don't need to assign null before reassigning to the correct value
3. The next line can fit on one line.